app/main.py

```diff
@@ -1,3 +1,20 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+from functools import wraps
+
+
+def cache(func: Callable) -> Callable:
+    dict__ = dict()
+
+    @wraps(func)
+    def inner(*args) -> Any:
+        nonlocal dict__
+
+        if not dict__.get((*args, inner.__name__,)) is None:
```

Comment:
As I see, here the cache key is a tuple of the function's arguments and the function name `(*args, inner.__name__,)`. This can be memory-inefficient if the function's arguments are large, as it creates a new tuple every time the function is called.

P.S It would be better to use `dict.get` method with a default value instead of using not `dict__.get((*args, inner.__name__,)) is None` to check if the key is present in the dictionary. This will simplify the code and make it easier to understand. Just saying ..

Reply:
Some cached result equals to zero that break some tests when 'if dict__.get(hash__):' used

Reply:
'''if dict__.get(hash__):'''